docs/guides/add-onboarding-flow.mdx

```diff
@@ -0,0 +1,245 @@
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+You will need a layout for the `/onboarding` route that will redirect users to the dashboard if they have already completed onboarding.
+
+1. In your `/app` directory, create an `/onboarding` folder.
+2.  In your `/onboarding` directory,  create a `layout.tsx` file and add the following code to the file. This file could also be expanded to handle multiple steps, if multiple steps are required for an onboarding flow.
+
+```tsx filename="src/app/onboarding/layout.tsx"
+import { auth } from "@clerk/nextjs/server";
+import { redirect } from "next/navigation";
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  if (auth().sessionClaims?.metadata.onboardingComplete === true) {
+    redirect("/dashboard");
+  }
+
+  return <>{children}</>;
+}
+```
+
+
+## Use `publicMetadata` to track user onboarding state
+
+Each Clerk user has a `User` object that contains a `publicMetadata` property, which can be used to store custom data about the user. This information can be accessed on the client side and can be used to drive application state. [Learn more about public metadata.](/docs/users/metadata#public-metadata)
+
+You can use the user's `publicMetadata` to track the user's onboarding state. To do this, you will create:
+
+- A process in your frontend with logic to collect and submit all the information for onboarding. In this guide, you will create a simple form.
+- A method in your backend to securely update the user's `publicMetadata`
+
+### Update the user's `publicMetadata` in your backend
+
+Now that there is a form to collect the user's onboarding information, you need to create a method in your backend to update the user's `publicMetadata` with this information. This method will be called when the user submits the form.
+
+1. In your `/onboarding` directory, create an `_actions.ts` file.
+2. Add the following code to the file. This file includes a method that will be called on form submission and will update the user's `publicMetadata` accordingly. The following example uses the [`clerkClient`](/docs/references/backend/overview) wrapper to interact with the Backend API and update the user's `publicMetadata`.
+
+```tsx
+"use server";
+
+import { auth, clerkClient } from "@clerk/nextjs/server";
+
+export const completeOnboarding = async (formData: FormData) => {
+  const { userId } = auth();
+
+  if (!userId) {
+    return { message: "No Logged In User" };
+  }
+
+  try {
+    const res = await clerkClient.users.updateUser(userId, {
+      publicMetadata: {
+        onboardingComplete: true,
+        applicationName: formData.get("applicationName"),
+        applicationType: formData.get("applicationType"),
+      },
+    });
+    return { message: res.publicMetadata };
+  } catch (err) {
+    return { error: "There was an error updating the user metadata." };
+  }
+};
+```
+
+
+## Collect user onboarding information
```

Comment:
This section still needs to get swapped with the section above "Update the user's `publicMetadata` in your backend".

This is because we list out what the user needs to do:
> - A process in your frontend with logic to collect and submit all the information for onboarding. In this guide, you will create a simple form.
> - A method in your backend to securely update the user's `publicMetadata`

and creating a form is first. so this section needs to go first